contracts/dao/ZDAO.sol

```diff
@@ -0,0 +1,252 @@
+import { GovernorSettings } from "@openzeppelin/contracts/governance/extensions/GovernorSettings.sol";
+import { GovernorCountingSimple } from "@openzeppelin/contracts/governance/extensions/GovernorCountingSimple.sol";
+import { GovernorVotes, IVotes } from "@openzeppelin/contracts/governance/extensions/GovernorVotes.sol";
+import {
+    GovernorTimelockControl,
+    TimelockController
+} from "@openzeppelin/contracts/governance/extensions/GovernorTimelockControl.sol";
+import {
+    GovernorVotesQuorumFraction
+} from "@openzeppelin/contracts/governance/extensions/GovernorVotesQuorumFraction.sol";
+import { GovernorPreventLateQuorum } from "@openzeppelin/contracts/governance/extensions/GovernorPreventLateQuorum.sol";
+
+// TODO dao: add missing NatSpec !!!
+/**
+ * @title ZDAO
+ * @notice A customizable governance contract based on OpenZeppelin's Governor contracts.
+ * @dev Extends OpenZeppelin's Governor contracts with various extensions for governance settings,
+ * voting, timelock control, and quorum fraction.
+ * See OpenZeppelin documentation: https://docs.openzeppelin.com/contracts/4.x/api/governance
+ * This is a fork of https://github.com/zer0-os/zDAO/blob/main/contracts/ZDAO.sol
+ * With updated parent contracts and Solidity version.
+ * @custom:security-contact admin@zer0.tech
+ */
+contract ZDAO is
+    Governor,
+    GovernorSettings,
+    GovernorCountingSimple,
+    GovernorVotes,
+    GovernorTimelockControl,
+    GovernorVotesQuorumFraction,
+    GovernorPreventLateQuorum
+{
+    /**
+     * @notice Creates a new ZDAO governance contract.
+     * @param governorName The name of the governor instance.
+     * @param token The governance token that allows users to vote.
+     * @param timelock The timelock controller that handles proposal execution delay.
+     * @param delay_ The delay before voting starts (in blocks).
+     * @param votingPeriod_ The duration of the voting period (in blocks).
+     * @param proposalThreshold_ The minimum number of votes required to create a proposal.
+     * @param quorumPercentage_ The quorum fraction required for a proposal to pass.
+     * @dev Initializes the governor with settings and extensions.
+     * See OpenZeppelin Governor documentation: https://docs.openzeppelin.com/contracts/4.x/api/governance
+     */
+    constructor(
+        string memory governorName,
+        IVotes token,
+        TimelockController timelock,
+        uint48 delay_,
+        uint32 votingPeriod_,
+        uint256 proposalThreshold_,
+        uint256 quorumPercentage_,
+        uint48 voteExtension_
+    )
+        Governor(governorName)
+        GovernorVotes(token)
+        GovernorTimelockControl(timelock)
+        GovernorSettings(
+            delay_,
+            votingPeriod_,
+            proposalThreshold_
+        )
+        GovernorVotesQuorumFraction(quorumPercentage_)
+        GovernorPreventLateQuorum(voteExtension_)
+    {}
+
+    function votingDelay()
```

Comment:
The indentation seems kind of weird and inconsistent for the functions in this file, some are one tab and some are two tabs?